parseAndPopulate/groupings.py

```diff
@@ -190,7 +194,14 @@ def _parse_and_load_not_api(self) -> tuple[int, int]:
                 except (ParseException, FileNotFoundError) as e:
                     self.log_module_creation_exception(e)
                     continue
+                if yang.organization != self.official_source and should_parse.seen_name_revision:
```

Comment:
shouldn't it be:
```
if self.official_source and yang.organization != self.official_source and should_parse.seen_name_revision:
```
as official_source can be None?

Reply:
`self.official_source` can, but `yang.organization` has type `str` so it can't be `None`.

Reply:
Oh but that should be fine. The logic here is supposed to be:
- if we haven't seen the name@revision before, we accept it from anywhere (`should_parse.seen_name_revision` will be false)
- otherwise, we accept it only if the this is the official source for the organization of that module (so only if `yang.organization == self.official_source`)

If neither of these is the case, both of the conditions in the `if` evaluate to true and we skip the module.

Reply:
understood, thanks